src/mame/skeleton/cubieboard4.cpp

```diff
@@ -0,0 +1,127 @@
+
+Main Cubieboard4 components:
+-Allwinner A80.
+-Realtek RTL8211E integrated Ethernet transceiver.
+-AMPAK Technology AP6330 WiFi + Bluetooth 4.0(HS) + FM Rx Module.
+-SKhynix KLM4G1YEMD-B031 (embedded MultiMediaCard Ver. 5.0 compatible, soldered to the back side of the PCB).
+-4 x H5TQ4G63AFR 4Gb DDR3 SDRAM.
+-X-Powers AXP809 PMIC.
+-X-Powers AC100 audio codec and RTC subsystem.
+
+***********************************************************************************************/
+
+#include "emu.h"
+#include "cpu/arm7/arm7.h"
+#include "bus/generic/slot.h"
+#include "bus/generic/carts.h"
+#include "softlist_dev.h"
+
+namespace {
+
+class cubiecca80_state : public driver_device
+{
+public:
+	cubiecca80_state(const machine_config &mconfig, device_type type, const char *tag) :
+		driver_device(mconfig, type, tag),
+		m_cart(*this, "cartslot"),
+		m_maincpu(*this, "maincpu")
+
+	{ }
+
+	void cubiecca80(machine_config &config);
+
+protected:
+	optional_device<generic_slot_device> m_cart;
+
+private:
+	required_device<cpu_device> m_maincpu;
+};
+
+static INPUT_PORTS_START( cubiecca80 )
+INPUT_PORTS_END
+
+void cubiecca80_state::cubiecca80(machine_config &config)
+{
+	// Basic machine hardware
+	ARM9(config, m_maincpu, 200'000'000); // Actually an Allwinner A80 2 GHz
+
+	// Video hardware
+	//SCREEN(...)
+
+	// Audio hardware
+	//SPEAKER(...)
+
+	GENERIC_CARTSLOT(config, m_cart, generic_plain_slot, "cubieboard4_sdcard");
+
+	// Software list for adding other compatible software (Linux distros, etc.).
+	SOFTWARE_LIST(config, "software_list").set_original("cubieboard4");
+}
+
+// Main machine, for booting SD card images from the software list
+
+ROM_START( cubieboard4 )
+ROM_END
+
+// Arcade games
+
+/* Monkey Jump. Android-based arcade from the Spanish company Falgas.
+   Has a separate I/O board, with a PIC18F46K22, connected by USB to the Cubieboard4 (using a RS-232 to USB adapter).
+   More info: https://www.recreativas.org/monkey-jump-4029-falgas */
+ROM_START( monkeyjmp )
+	DISK_REGION( "nand" )
+	DISK_IMAGE( "mmcblk1", 0, SHA1(5c005005b2ca17b916b2cccd48d291b19337d4cc) )
+
+	ROM_REGION( 0x2000, "io", 0 )
+	ROM_LOAD( "pic18f46k22.bin", 0x0000, 0x2000, NO_DUMP ) // 1024 bytes internal ROM
+ROM_END
+
+// Other "official" software (from cubieboard.org) for the internal eMMC
+
+ROM_START( cb4_android41 )
+	DISK_REGION( "nand" )
+	DISK_IMAGE( "android4.4-cb4-emmc-v4.1.20161119", 0, SHA1(ccf854e17936e3c57a1701f9edb3c63b704d4d59) )
+ROM_END
+
+ROM_START( cb4_android43 )
+	DISK_REGION( "nand" )
+	DISK_IMAGE( "android4.4-cb4-emmc-v4.3.20170717", 0, SHA1(bb386ce76686698ad7f25b0469970c58abf4745e) )
+ROM_END
+
+ROM_START( cb4_debiansvr )
+	DISK_REGION( "nand" )
+	DISK_IMAGE( "cb4-debian-server-hdmi-emmc-v1.0",  0, SHA1(31db5866016cfd6f1be331c70859cdb27022a7aa) )
+ROM_END
+
+ROM_START( cb4_linarodst )
+	DISK_REGION( "nand" )
+	DISK_IMAGE( "linaro-desktop-cb4-emmc-hdmi-v1.1", 0, SHA1(249095e6faaee11e330eaf59285c2e68d0b5a30f) )
+ROM_END
```

Comment:
Not sure I gather this: what are those romset wrt what's available in SW list?

Reply:
There are two kind of images, ones to be restored on a SD and then inserted on the MicroSD slot (on the software list) and others to be transferred to the internal NAND Flash (on the driver).
The internal NAND has a eMMC interface, but is not removable media (flash chips directly soldered to the PCB).

Should we move eMMC images to the software list also, creating a new interface (something like "cubieboard4_emmc"), and leaving on the driver just the arcade games?

Thanks!!!!

Reply:
Inclined to think that they both belongs to the same SW items, as different interface loading.